peak/mapper/binding.py

```diff
@@ -0,0 +1,251 @@
+import typing as tp
+import itertools as it
+from hwtypes import SMTBit, SMTBitVector, SMTSIntVector, AbstractBitVector, AbstractBit
+from hwtypes import is_adt_type
+from hwtypes.adt import Product, Enum, Sum
+from .util import SubTypeDict
+
+
+__ALL__ = ['Binder', 'get_from_path', 'set_from_path', 'binding_pretty_print']
+
+def copy_smt_value(val):
+    if isinstance(val,(SMTBit,SMTBitVector)):
+        return type(val)(val.value)
```

Comment:
This doesn't do anything meaningful unless you are going to be setting attributes on val.

Reply:
Fixed by #76